### PR TITLE
Top level span provides information about the request being made

### PR DESCRIPTION
**Pull request by ferozco**
## Before this PR
The traces we produced where not very helpful since they did not show you what request was being made.

image

## After this PR
==COMMIT_MSG==
Top level span of a request contains the method, service and endpoint that is being called
==COMMIT_MSG==
image

## Possible downsides?
N/A

### CODE REVIEW

File: dialogue-core/src/main/java/com/palantir/dialogue/core/DialogueTracedRequestChannel.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    private final Channel delegate;
+
+    public DialogueTracedRequestChannel(Channel delegate) {
+        this.delegate = delegate;
+    }
+
+    @Override
+    public ListenableFuture<Response> execute(Endpoint endpoint, Request request) {
+        String operationName =
+                "Dialogue: " + endpoint.httpMethod() + " " + endpoint.serviceName() + "#" + endpoint.endpointName();
```
Comment (carterkozak) [inline dialogue-core/src/main/java/com/palantir/dialogue/core/DialogueTracedRequestChannel.java]: I don’t think we need the http method here, either http method + template, or service + endpoint name should be enough uniquely identidying data

Comment (ferozco) [reply dialogue-core/src/main/java/com/palantir/dialogue/core/DialogueTracedRequestChannel.java]: I thought it was nice to provide as much information as possible, but I can remove the method if you want

**Review by carterkozak — APPROVED** [review]

File: dialogue-core/src/main/java/com/palantir/dialogue/core/DialogueTracedRequestChannel.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    private final Channel delegate;
+
+    public DialogueTracedRequestChannel(Channel delegate) {
+        this.delegate = delegate;
+    }
+
+    @Override
+    public ListenableFuture<Response> execute(Endpoint endpoint, Request request) {
+        String operationName = "Dialogue: request " + endpoint.serviceName() + "#" + endpoint.endpointName();
+        return Tracers.wrapListenableFuture(operationName, () -> delegate.execute(endpoint, request));
```
Comment (iamdanfox) [inline dialogue-core/src/main/java/com/palantir/dialogue/core/DialogueTracedRequestChannel.java:36]: @ferozco I think previously we included the HTTP verb too, currently it's just the service & endpoint

Comment (carterkozak) [reply dialogue-core/src/main/java/com/palantir/dialogue/core/DialogueTracedRequestChannel.java:36]: https://github.com/palantir/dialogue/pull/582#discussion_r399687592

Comment (iamdanfox) [reply dialogue-core/src/main/java/com/palantir/dialogue/core/DialogueTracedRequestChannel.java:36]: fair enough

**Pull request merged**